### PR TITLE
Cron jobs for certificate

### PR DESCRIPTION
**Pull request by SirAndrii**
## Description

A cron job Was written for certificates (link to iisue https://github.com/ita-social-projects/horondi_client_be/issues/1035):

- [x] remove from DB certificates that are older than three years
- [x] change status for Expired certificates
- [x] send reminders for certificates that are expiring in the next 30 days

Tests with mocking modules
image

Tests were made locally with changing cron period and comparing changes in DB 

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: helpers/cron-job/unlock-users.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ const unlockUsers = () =>
             afterChanges
           );
           await addHistoryRecord(historyRecord);
-          return sendEmail(userData.email, UNLOCK_USER);
```
Comment (SirAndrii) [inline helpers/cron-job/unlock-users.js:70]: We are in for loop where each user that needs to be unlocked is iterating. Using return will break the loop, so I removed it.

File: helpers/cron-job/certificates-expire-check.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      },
+    };
+
+    await CertificateModel.updateMany(expiredFilter, {
+      isActive: false,
+    }).exec();
+
+    const reminderFilter = {
+      isUsed: false,
+      isActive: true,
```
Comment (SirAndrii) [inline helpers/cron-job/certificates-expire-check.js:42]: Maybe we should rename this field to isExpired?

File: helpers/cron-job/unlock-users.js (hunk 1 above)
Comment (AndriyLymych) [reply helpers/cron-job/unlock-users.js:70]: use Promise.all and map

Comment (SirAndrii) [reply helpers/cron-job/unlock-users.js:70]: done

File: utils/modify-date.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const _ = undefined;
```
Comment (VasylLymych) [inline utils/modify-date.js:1]: Why is here such a strange constant? Is it exists only for test above? If that so than why don't you just write modifyNowDate(null, null, 10) for example?

File: tests/helpers/cron/certificates-expire-check.test.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+describe('Cron job check', () => {
+  it('Mongo methods and emailSender were called', async () => {
+    await certificatesExpireCheck();
+
+    expect(isCalledManyUpdate).toBeTruthy();
+
+    expect(isCalledFind).toBeTruthy();
+
+    expect(isCalledSender).toBeTruthy();
```
Comment (VasylLymych) [inline tests/helpers/cron/certificates-expire-check.test.js:41]: I would do it like: expect([isCalledManyUpdate, isCalledFind, isCalledSender].every(el => el)).toBeTruthy()

File: utils/modify-date.js (hunk 3 above)
Comment (SirAndrii) [reply utils/modify-date.js:1]: Not just for tests. This function can be reused in other parts of code. There is a choice, write modifyNowDate( undefined, undefined, 10 ) or shortener - modifyNowDate(  _, _, 10 ) . I don't know which way is better.

**Review by maarekua — APPROVED** [review]

**Review by boris-giga — APPROVED** [review]

**Pull request merged**